yt_dlp/extractor/niconico.py

```diff
@@ -367,229 +388,120 @@ def parse_format_id(id_code):
-                    'url': video_real_url,
-                    'format_id': 'smile' if not is_economy else 'smile_low',
-                    'format_note': 'SMILEVIDEO source' if not is_economy else 'SMILEVIDEO low quality',
-                    'ext': extension,
-                    'container': extension,
-                    'vcodec': v_stream.get('codec_name'),
-                    'acodec': a_stream.get('codec_name'),
-                    # Some complex swf files doesn't have total bit rate metadata (e.g. nm6049209)
-                    'tbr': int_or_none(metadata['format'].get('bit_rate'), scale=1000),
-                    'vbr': int_or_none(v_stream.get('bit_rate'), scale=1000),
-                    'abr': int_or_none(a_stream.get('bit_rate'), scale=1000),
-                    'height': int_or_none(v_stream.get('height')),
-                    'width': int_or_none(v_stream.get('width')),
-                    'source_preference': 5 if not is_economy else -2,
-                    'quality': 5 if is_source and not is_economy else None,
-                    'filesize': filesize
-                })
+        quality_info = api_data['media']['delivery']['movie']
+        session_api_data = quality_info['session']
+        for (audio_quality, video_quality, protocol) in itertools.product(quality_info['audios'], quality_info['videos'], session_api_data['protocols']):
+            fmt = self._extract_format_for_quality(video_id, audio_quality, video_quality, protocol)
+            if fmt:
+                formats.append(fmt)
 
         self._sort_formats(formats)
 
         # Start extracting information
         title = (
-            get_video_info_xml('title')  # prefer to get the untranslated original title
-            or get_video_info_web(['originalTitle', 'title'])
-            or self._og_search_title(webpage, default=None)
-            or self._html_search_regex(
-                r'<span[^>]+class="videoHeaderTitle"[^>]*>([^<]+)</span>',
-                webpage, 'video title'))
-
-        watch_api_data_string = self._html_search_regex(
-            r'<div[^>]+id="watchAPIDataContainer"[^>]+>([^<]+)</div>',
-            webpage, 'watch api data', default=None)
-        watch_api_data = self._parse_json(watch_api_data_string, video_id) if watch_api_data_string else {}
-        video_detail = watch_api_data.get('videoDetail', {})
-
-        thumbnail = (
-            self._html_search_regex(r'<meta property="og:image" content="([^"]+)">', webpage, 'thumbnail data', default=None)
-            or dict_get(  # choose highest from 720p to 240p
-                get_video_info_web('thumbnail'),
-                ['ogp', 'player', 'largeUrl', 'middleUrl', 'url'])
-            or self._html_search_meta('image', webpage, 'thumbnail', default=None)
-            or video_detail.get('thumbnail'))
-
-        description = get_video_info_web('description')
+            get_video_info(['originalTitle', 'title'], get_first=True)
```

Comment:
```suggestion
            get_video_info(('originalTitle', 'title'))
```
`get_first=True` is already the default